selftests/functional/test_basic.py

```diff
@@ -105,7 +107,11 @@ def probe_binary(binary):
 
 
 CC_BINARY = probe_binary('cc')
+# On macOS, the default coreutils installation (brew)
+# installs the gnu utility versions with a g prefix.
 ECHO_BINARY = probe_binary('echo')
+if ECHO_BINARY is not None:
+    ECHO_BINARY = probe_binary('gecho')
```

Comment:
This second probe should be `find_command` to catch issue where `echo` nor `gecho` are available.